project1/application.py

```diff
@@ -1,75 +1,92 @@
-from sqlalchemy.ext.declarative import declarative_base
-#import sqlalchemy as db
-from werkzeug.security import generate_password_hash
 from datetime import datetime
+from flask import flash, Flask, session, render_template, request, redirect, url_for
+from werkzeug.security import generate_password_hash
+from flask_session import Session
+from sqlalchemy.ext.declarative import declarative_base
 
-Base = declarative_base()
-app = Flask(__name__)
-
-class User(Base):
-    __tablename__ = 'users'
-    id = Column(Integer, Sequence('user_id_seq'),primary_key=True)
-    username = Column(String,primary_key=True)
-    password = Column(String)
-    created_time = Column(DateTime)
-
-    def hash_password(self, password):
-        self.password = generate_password_hash(password)
+from models import User
 
+BASE = declarative_base()
+APP = Flask(__name__)
 
 # Check for environment variable
 if not os.getenv("DATABASE_URL"):
     raise RuntimeError("DATABASE_URL is not set")
 
 # Configure session to use filesystem
 
-app.config["SESSION_PERMANENT"] = False
-app.config["SESSION_TYPE"] = "filesystem"
-Session(app)
+APP.config["SESSION_PERMANENT"] = False
+APP.config["SESSION_TYPE"] = "filesystem"
+Session(APP)
 
 # Set up database
-engine = create_engine("postgres://onucnnqeicevfn:c2b89ff6bebb4f290d6b7d5cc6e1ac1ff55b1cee18da41f3145983e36f0cc838@ec2-54-197-48-79.compute-1.amazonaws.com:5432/d8aehqfbjgde2h")
-Base.metadata.create_all(engine)
-db = scoped_session(sessionmaker(bind=engine))
-session=db()
+ENGINE = create_engine(os.getenv("DATABASE_URL"))
+BASE.metadata.create_all(ENGINE)
+DB = scoped_session(sessionmaker(bind=ENGINE))
+DB_SESSION = DB()
 
-@app.route("/")
+@APP.route("/")
 def index():
     return "Project 1: TODO"
 
 
-@app.route("/register",methods = ['POST','GET'])
+@APP.route("/register", methods=['POST', 'GET'])
 def register():
     if request.method == 'POST':
-       name = request.form['uname']
-       print(name)
-       password = request.form['password']
-       
-       r=User(username=name,created_time=datetime.utcnow())
-       r.hash_password(password)
-       try:
-       	   print("saved record")
-       	   if session.query(exists().where(User.username == name)).scalar():
-       	      return render_template('result.html',value="Already registered")
-           else:
-           		db.add(r)
-           		db.commit()
-           		return render_template('result.html',value="successfully registered")
-       except Exception as e:
-           print(e)
-           return render_template('error.html')
+        name = request.form['uname']
+        print(name)
+        password = request.form['password']
+        print(password)
+        record = User(username=name, password=password, created_time=datetime.utcnow())
+      # r.hash_password(password)
+        try:
+            print("saved record")
+            if DB_SESSION.query(exists().where(User.username == name)).scalar():
```

Comment:
how about avoid a call to DB here to check whether the user is already registered or not. You can do so this by making use of exceptions